test/test.js

```diff
@@ -587,6 +587,39 @@ describe('Page', function() {
     }));
   });
 
+  describe('Page.getMetrics', function() {
+    it('should get metrics from a page', SX(async function() {
+      await page.goto('about:blank');
+      const metrics = await page.getMetrics();
+      checkMetrics(metrics);
+    }));
+    it('metrics event fired on console.timeStamp', SX(async function() {
+      const metricsPromise = new Promise(fulfill => page.once('metrics', fulfill));
+      await page.evaluate(() => console.timeStamp('test42'));
+      const metrics = await metricsPromise;
+      expect(metrics.title).toBe('test42');
+      checkMetrics(metrics.metrics);
+    }));
+    function checkMetrics(metrics) {
+      expect(metrics['Timestamp']).toBeGreaterThan(0);
```

Comment:
can we also check that there are no other fields beyond these? this would allow us to update docs when something new is added

Reply:
There are other fields coming over the protocol. But we don't want to document/expose those in puppeteer.